save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/controllers/DownloadProjectController.kt

```diff
@@ -105,21 +130,57 @@ class DownloadProjectController(private val configProperties: ConfigProperties)
         }
     }
 
+    private fun saveBinaryFile(
+        binaryExecutionRequest: BinaryExecutionRequest,
+        property: File,
+        binFile: File,
+    ) {
+        val tmpDir = generateDirectory(binFile.name.hashCode(), binFile.name)
+        val pathToProperties = tmpDir.path + File.separator + property.name
+        property.copyTo(File(pathToProperties))
+        binFile.copyTo(File(tmpDir.path + File.separator + binFile.name))
+        val project = binaryExecutionRequest.project
+        sendToBackendAndOrchestrator(
+            project,
+            pathToProperties,
+            tmpDir.relativeTo(File(configProperties.repository)).normalize().path,
+            ExecutionType.STANDARD,
+            binaryExecutionRequest.testsSuites.map { TestSuiteDto(TestSuiteType.STANDARD, it, project, pathToProperties) }
+        )
+    }
+
+    private fun generateDirectory(hashName: Int, dirName: String): File {
+        val tmpDir = File("${configProperties.repository}/$hashName")
+        if (tmpDir.exists()) {
+            tmpDir.deleteRecursively()
+            log.info("For $dirName file: dir $hashName was deleted")
+        }
+        tmpDir.mkdirs()
+        log.info("For $dirName repository: dir $hashName was created")
+        return tmpDir
+    }
+
     @Suppress(
         "LongMethod",
         "ThrowsCount",
         "TooGenericExceptionCaught",
         "TOO_LONG_FUNCTION",
         "LOCAL_VARIABLE_EARLY_DECLARATION")
-    private fun sendToBackendAndOrchestrator(project: Project, propertiesRelativePath: String, resourcesRootPath: String) {
+    private fun sendToBackendAndOrchestrator(
+        project: Project,
+        propertiesRelativePath: String,
+        resourcesRootPath: String,
+        executionType: ExecutionType,
+        testSuitesDto: List<TestSuiteDto>?
```

Comment:
Please add a precondition, that we should only have not null `testSuiteDtos` if `executionType == STANDATD`